sdk/src/types/api/core/response.rs

```diff
@@ -151,22 +194,15 @@ pub enum LedgerInclusionState {
 )]
 pub struct BlockMetadataResponse {
     pub block_id: BlockId,
-    pub parents: Vec<BlockId>,
-    pub is_solid: bool,
-    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
-    pub referenced_by_milestone_index: Option<u32>,
-    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
-    pub milestone_index: Option<u32>,
-    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
-    pub ledger_inclusion_state: Option<LedgerInclusionState>,
+    // TODO: verify if really optional: https://github.com/iotaledger/tips-draft/pull/24/files#r1293426314
```

Comment:
I wonder why `tx_state` is also optional? why `block_state` should be but `tx_state` not?

Reply:
Because there are blocks without a transaction payload